helm/resource_release.go

```diff
@@ -840,15 +840,18 @@ func resourceDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{})
 	if d.HasChanges(recomputeMetadataFields...) {
 		d.SetNewComputed("metadata")
 	}
-	if d.HasChange("version") {
-		// only recompute metadata if the version actually changes
-		// chart versioning is not consistent and some will add
-		// a `v` prefix to the chart version after installation
-		old, new := d.GetChange("version")
-		oldVersion := strings.TrimPrefix(old.(string), "v")
-		newVersion := strings.TrimPrefix(new.(string), "v")
-		if oldVersion != newVersion {
-			d.SetNewComputed("metadata")
+	repo := d.Get("repository").(string)
+	if !isLocalChart(d.Get("chart").(string), repo) || registry.IsOCI(repo) {
```

Comment:
Is there a way we could make the check that it's an OCI part of the isLocalChart function?